tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ThreadContextTest.java

```diff
@@ -582,7 +583,231 @@ public void thirdPartyContextProvidersAreIncludedInThreadContext() {
+                        "Context type was not propagated to contextual action.");
+
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type that is configured to be cleared was not cleared.");
+
+                Assert.assertEquals(Thread.currentThread().getId(), testThreadId,
+                        "Completion stages created via withContextCapture must run on the test case's main thread " +
+                        "because it both completes the original stage and requests the result. The withContextCapture " +
+                        "method does not allow new async threads to be allocated even when the *Async methods are used.");
+
+                Label.set("withContextCapture-CompletableFuture-test-label-D");
+
+                return i + i;
+            });
+
+            // Original stage remains usable, but not having been created by a ManagedExecutor, does not make any
+            // guarantees about context propagation
+            CompletableFuture<Integer> unmanagedStage5 = unmanagedStage1.thenApply(i -> i / 2);
+
+            Label.set("withContextCapture-CompletableFuture-test-label-E");
+
+            unmanagedStage1.complete(1010);
+
+            Assert.assertEquals(stage2.getNow(9090), Integer.valueOf(1010),
+                    "Completion stage created by withContextCapture did not complete with same value as original stage.");
+
+            Assert.assertEquals(stage3.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(2020),
+                    "Incorrect or missing result of completion stage.");
+
+            Assert.assertEquals(stage4.join(), Integer.valueOf(4040),
+                    "Incorrect or missing result of completion stage.");
+
+            Assert.assertEquals(unmanagedStage5.get(), Integer.valueOf(505),
+                    "Incorrect or missing result of completion stage.");
+
+            // Has context been properly restored after the contextual operation(s)?
+            Assert.assertEquals(Buffer.get().toString(), "withContextCapture-CompletableFuture-test-buffer",
+                    "Previous context was not restored after context was cleared for contextual action.");
+            Assert.assertEquals(Label.get(), "withContextCapture-CompletableFuture-test-label-E",
+                    "Previous context was not restored after context was propagated for contextual action.");
+        }
+        finally {
+            // Restore original values
+            Buffer.set(null);
+            Label.set(null);
+        }
+    }
+
+    /**
+     * Verify that the MicroProfile Concurrency ThreadContext implementation's withContextCapture
+     * method can be used to create a dependent CompletableFuture instance that completes when the
+     * original stage completes and runs dependent stage actions with context that is captured
+     * from the thread that creates the dependent stage.
+     */
```

Comment:
Should the test description talk about testing a dependent CompletionStage rather than CompletableFuture?

Reply:
Yes, that needs to be updated